back/boxtribute_server/business_logic/statistics/crud.py

```diff
@@ -475,3 +487,42 @@ def compute_stock_overview(base_id):
         "size", "location", "category", "tag", "dimension", facts=facts
     )
     return {"facts": facts, "dimensions": dimensions}
+
+
+def create_shareable_link(
+    *, user_id, base_id, view, valid_until=None, url_parameters=None
+):
+    """Insert information for a new shareable link. Create unique 8-digit code."""
+    now = utcnow()
+    if valid_until is not None:
+        if valid_until.tzinfo is None:
+            # If valid_until doesn't have any tzinfo, interprete it as UTC
+            valid_until = valid_until.replace(tzinfo=dtimezone.utc)
+        else:
+            valid_until = valid_until.astimezone(dtimezone.utc)
+        if valid_until < now:
+            return InvalidDate(date=valid_until)
+
+    link = None
+    for _ in range(RANDOM_SEQUENCE_GENERATION_ATTEMPTS):
```

Comment:
I'd suggest moving this to a hash function in order to get rid of the error handling, but can life without it.